src/shadow.rs

```diff
@@ -335,8 +335,8 @@ struct ContainsSelf {
 }
 
 impl<'v> Visitor<'v> for ContainsSelf {
-    fn visit_ident(&mut self, _: Span, ident: Ident) {
-        if self.name == ident.unhygienic_name {
+    fn visit_name(&mut self, _: Span, name: Name) {
+        if self.name == name {
```

Comment:
If some lints don't work due to gensyms after proper replacement of all `unhygienic_name`s with `unhygienize()` , they need to be moved to AST.